app/Utils/Traits/GeneratesCounter.php

```diff
@@ -583,8 +583,16 @@ private function resetCounters(Client $client)
         $settings->recurring_expense_number_counter = 1;
         $settings->purchase_order_number_counter = 1;
 
+        if( config('ninja.reset_all_counter') ) {
+            $settings->client_number_counter = 1;
+        }
+
         $client->company->settings = $settings;
         $client->company->save();
+
+        if ($new_reset_date->lte(now())) {
+            return $this->resetCounters($client);
+        }
```

Comment:
This is dangerous. It assumes that the current client number pattern is suitable for the counter to be reset. There cannot be duplicates, so you would need to have a pattern which allows this to function.

Reply:
Per default this is disabled. But enabling this will not break anything because you're already checking for exisiting ids in `->checkEntityNumber()`. In first MR I also added a small performance improvement to get latest ID instead of starting with 1 & increase in loop, but this was too risky for you to change (fully acceptable! :-)), but this should also not break anything.

If someone is enabling this setting, normally he knows what he wants and what he may get. If client pattern may leads to "conflicts"/unprettier ids (currently: loop `checkEntityNumber()` 100 times, after that don't increase id, instead generate a random one -> a bit unprettier IDs), its just because the pattern is not unique enough (e.g. not on daily date base / monthly date base with monthly reset intervall...).